### ISSUE TITLE
FF: cck2 causes error when updating to 58

### ISSUE THREAD

**Issue by star-buck** [blue-systems/netrunner-issues-old#26]
https://github.com/netrunner-rolling/QA-Bugs/issues/18#issuecomment-362368073

**Comment by star-buck** [comment]
Any idea how to fix this without requiring anything special from the user side?

**Comment by NuLogicSystems** [comment]
I just posted this to netrunner-releases, it needs to be fixed in cck2 itself.
https://github.com/blue-systems/netrunner-releases/issues/25

**Comment by NuLogicSystems** [comment]
Once they fix it in netrunner-common, I can fix in netrunner-rolling.

**Comment by NuLogicSystems** [comment]
Opps, it looks like we both posted this here, I'll close mine.

**Comment by shadeslayer** [comment]
Hi
It seems like this is directly related to https://github.com/mkaply/cck2wizard/issues/75

Could you grab this https://raw.githubusercontent.com/mkaply/cck2wizard/master/cck2/modules/Preferences.jsm and place it in /usr/lib/firefox/cck2/modules/Preferences.jsm to see if it fixes your problem?

**Comment by NuLogicSystems** [comment]
Nope, that did not help. I'm still getting the exact same error.

**Comment by shadeslayer** [comment]
Ah damn, is it the same error? Could you perhaps follow up on the cc2kwizard ticket if it is?

**Comment by NuLogicSystems** [comment]
That thread says something about running a script to clean out or delete the startupCache.
There is also another version of this file ziped up and linked to in a later post from the file you linked to.

**Comment by shadeslayer** [comment]
I presumed you were starting from a empty ~/.mozilla

**Comment by NuLogicSystems** [comment]
No, even the netrunner-common firefox settings package doesn't have an empty ~/.mozilla folder, but that's not the issue.

After I maually delete my ~/.cache/mozilla folder your file worked.
However, it would be better if the end user doesn't need to do this.

**Comment by shadeslayer** [comment]
Yes, we don't have a empty ~/.mozilla as much as I'd like to :(

**Comment by NuLogicSystems** [comment]
The .mozilla folder isn't the problem, it the startupcache folder that is (~/.cache/mozilla).

**Comment by NuLogicSystems** [comment]
Would also changing the "cckVersion": "2.2.3.4" string in cck2.cfg file to  "cckVersion":  "2.2.3.5" triger a startupcahe update?

**Comment by NuLogicSystems** [comment]
Update: Changing the cck2 version string seems to have fixed the startupcache issue with the updated Preferences.jsm file on my test machine.
@star-buck, could you please update to the new netrunner-settings-firefox package I just pushed to the "netrunner" repo for rolling?

**Comment by shadeslayer** [comment]
@NuLogicSystems I'd recommend you ship a entirely new CCK2 instead of only changing certain parts.

**Comment by NuLogicSystems** [comment]
@shadeslayer I don't have FF ESR to run the wizard on, or is it just replacing the modules dir and updateing the version like last time?

**Comment by shadeslayer** [comment]
It should be as easy as just unpacking the new release over the old one, yes

**Comment by NuLogicSystems** [comment]
Cool, I need to keep that in mind.

**Comment by star-buck** [comment]
How to prevent any breakage in the future?

**Comment by NuLogicSystems** [comment]
There really wasn't many more changes other than the Preferences.jsm file, but I went ahead and updated the other two files as well.
https://github.com/netrunner-rolling/netrunner-settings-firefox/commit/6c9e6081067429f9e7c31caf09a11c1a7d285342

**Comment by NuLogicSystems** [comment]
@star-buck I'd imagine we will need to monitor cck2 for issues and/or updates continually.

**Comment by shadeslayer** [comment]
@star-buck that entirely depends on Firefox TBH, if they change something in their API to configure settings, then CCK2 of course breaks.

**Comment by NuLogicSystems** [comment]
@shadeslayer  It looks like FF is also dropping distrobution/bundles soon, if not already.
This may be the next area of concern for me, I use distrobution filder to ship my extensions.
https://mike.kaply.com/2015/04/07/the-future-of-cck2/

Following the link there, I would need to move them to cck2's bundles folder.
However, I'm not exactly 100% sure where to put the bundles foler in cck2 and/or if there is anything else I would need to edit for this.

For now using the /usr/lib/firefox/distrobution/extensions folder is still working.

**Comment by NuLogicSystems** [comment]
Removed my last post.
That thread I was concerned about was over 2 years old and not actually effecting me.

**Comment by star-buck** [comment]
Lets make sure this doesnt hit us again next update... also is this fixed
by simply updating our packaging and people continue or help thrmselves via
a simple upgrade (not a special upgrade command, just a normal update
routine via octopi or discover?)

**Comment by NuLogicSystems** [comment]
Since we are updating all of cck2 and not just the Preferences.jsm file, a normal update should be all that is needed.

**Comment by shadeslayer** [comment]
The best way to make sure we don't hit this again in the future is to write a QA test that ensures that everytime we update Firefox, we ensure that the settings are applied on a clean system.

**Comment by NuLogicSystems** [comment]
What do you mean by clean system?
Manjaro pulls in Arches Firefox package minus the branding. Now that I know we need to update cck2 itself more often, I'll be looking out for that. I'll also need to test FF out more often in the Manjaro testing branch before it hits stable. My build system does run on Manjaro testing, but I don't typically use FF. I only caught it because of the pre-load on one of my test systems.

**Comment by shadeslayer** [comment]
Should this be closed?
